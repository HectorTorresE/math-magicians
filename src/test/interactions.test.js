import { BrowserRouter } from 'react-router-dom';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NavBar from '../components/navbar';

describe('Navigation interaction', () => {
  test('Home', () => {
    const Element = (
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    );
    render(Element);
    const home = screen.getByText('Home');
    userEvent.click(home);

    expect(home.classList.contains('active')).toBeTruthy();
  });
  test('Calculator', () => {
    const Element = (
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    );
    render(Element);
    const calculator = screen.getByText('Calculator');
    userEvent.click(calculator);

    expect(calculator.classList.contains('active')).toBeTruthy();
  });
  test('Quote', () => {
    const Element = (
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    );
    render(Element);
    const quote = screen.getByText('Quote');
    userEvent.click(quote);
    expect(quote.classList.contains('active')).toBeTruthy();
  });
});
