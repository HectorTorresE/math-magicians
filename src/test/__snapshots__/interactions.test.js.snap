// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components testing Calculator Component 1`] = `
<div
  className="calculator"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="result"
  >
    0
  </div>
  <div
    className="digits"
  >
    <button
      type="button"
    >
      AC
    </button>
    <button
      type="button"
    >
      +/-
    </button>
    <button
      type="button"
    >
      %
    </button>
    <button
      className="operators"
      type="button"
    >
      รท
    </button>
    <button
      type="button"
    >
      7
    </button>
    <button
      type="button"
    >
      8
    </button>
    <button
      type="button"
    >
      9
    </button>
    <button
      className="operators"
      type="button"
    >
      *
    </button>
    <button
      type="button"
    >
      4
    </button>
    <button
      type="button"
    >
      5
    </button>
    <button
      type="button"
    >
      6
    </button>
    <button
      className="operators"
      type="button"
    >
      -
    </button>
    <button
      type="button"
    >
      1
    </button>
    <button
      type="button"
    >
      2
    </button>
    <button
      type="button"
    >
      3
    </button>
    <button
      className="operators"
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      type="button"
    >
      0
    </button>
    <button
      type="button"
    >
      .
    </button>
    <button
      className="operators"
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Components testing NavBar Component 1`] = `
<nav
  className="NavBar"
>
  <img
    alt="Math"
    src="logo.png"
  />
  <h1>
    Math Magicians
  </h1>
  <div>
    <a
      aria-current="page"
      className="link active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "isActive": "active-link",
        }
      }
    >
      Home
    </a>
    <a
      className="link"
      href="/calculator"
      onClick={[Function]}
      style={
        Object {
          "isActive": "active-link",
        }
      }
    >
      Calculator
    </a>
    <a
      className="link"
      href="/quote"
      onClick={[Function]}
      style={
        Object {
          "isActive": "active-link",
        }
      }
    >
      Quote
    </a>
  </div>
</nav>
`;
